packages/interactivity/src/vdom.js

```diff
@@ -118,14 +118,20 @@ export function toVdom( root ) {
 			);
 		}
 
-		let child = treeWalker.firstChild();
-		if ( child ) {
-			while ( child ) {
-				const [ vnode, nextChild ] = walk( child );
-				if ( vnode ) children.push( vnode );
-				child = nextChild || treeWalker.nextSibling();
+		if ( node.localName === 'template' ) {
+			props.content = [ ...node.content.childNodes ].map( ( childNode ) =>
+				toVdom( childNode )
+			);
```

Comment:
It's not very important but, can't we do `toVdom( node.content )` to create a single `treeWalker`? Or `node.content` is not a proper node? Maybe doing it that way creates an extra top node that we don't want?

Reply:
> Or `node.content` is not a proper node?

I'm afraid [`HTMLTemplateElement.content`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLTemplateElement/content) is not part of the document tree; it is a separate [`DocumentFragment`](https://developer.mozilla.org/en-US/docs/Web/API/DocumentFragment).

We can only traverse it with a new `TreeWalker` instance. 🤷

Reply:
Sorry, I think I've misunderstood your question, haha. 😄 I thought you were talking about reusing the current `TreeWalker` instance. Forget my previous message!

Answering to your actual question: [`HTMLTemplateElement.content`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLTemplateElement/content) is a [DocumentFragment](https://developer.mozilla.org/en-US/docs/Web/API/DocumentFragment), not a `Node`, so we can't call `toVdom( node.content )`.

We would have to (1) create a new node, (2) add the document's content there, then (3) process the node, and then (4) get the node children.

For me, it was simpler to map through `node.content.childNodes`.

Reply:
Yep, cool then. Thank you 🙂